gse-demo/src/main/java/com/powsybl/gse/demo/GseDemo.java

```diff
@@ -9,18 +9,102 @@
 import com.powsybl.gse.app.GseApp;
 import org.slf4j.bridge.SLF4JBridgeHandler;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.PrintWriter;
+
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 public final class GseDemo {
 
+    private static final String UNCAUGHT_EXCEPTION_MESSAGE = "FATAL: uncaught throwable in javafx startup";
+    private static final String STARTUP_ERROR_LOGFILE_NAME = "GSE_STARTUP_ERROR.log";
+
     private GseDemo() {
     }
 
+    //The system is utterly broken. Try as many things as we can to get some info to the user.
+    private static void lastResortFallbackErrorLog(Throwable t) {
+        //We don't even have a logging system, this happens for example
+        //if we are missing jars or if the magic javalauncher has a problem.
+        //Try to show the error everywhere so that the user has the best chance of noticing it...
+        try {
+            //Try stdout
+            System.out.println(UNCAUGHT_EXCEPTION_MESSAGE);
+            t.printStackTrace();
+        } catch (Throwable throwableSystemOut) {
+            //Swallow throwableSystemOut here, we don't want to confuse the user.
+            //This is such a broken case that we don't want to
+            //log throwableSystemOut, we want the other mechanisms in this method
+            //to log the rootThrowable to fix the root cause.
```

Comment:
In what case a throwable is thrown? An `OutOfMemoryError` or if the hard-drive if full?

Reply:
I'm not sure under what circumstances System.out.println can throw. Certainly in very broken cases that are hard to predict, which is when we need logging the most to understand the situation. Also this adds symmetry to the other attempts at logging (which can fail more predictably, for example because of file permissions)